internal/util/util.go

```diff
@@ -60,15 +65,51 @@ func GetTempDir() string {
 }
 
 // ExportFromReader reads bytes from given reader and exports to external tar, directory or stdout.
-func ExportFromReader(input io.Reader, opts define.BuildOutputOption) error {
+func ExportFromReader(input io.ReadCloser, store storage.Store, opts define.BuildOutputOption) error {
 	var err error
 	if !filepath.IsAbs(opts.Path) {
 		opts.Path, err = filepath.Abs(opts.Path)
 		if err != nil {
 			return err
 		}
 	}
-	if opts.IsDir {
+	if opts.Type == "image" {
+		if opts.Attrs["push"] != "true" {
+			return nil
+		}
+
+		image := opts.Attrs["name"]
+		destSpec := opts.Attrs["name"]
+		dest, err := alltransports.ParseImageName(destSpec)
+		// add the docker:// transport to see if they neglected it.
+		if err != nil {
+			destTransport := strings.Split(destSpec, ":")[0]
+			if t := transports.Get(destTransport); t != nil {
+				return err
+			}
+
+			if strings.Contains(destSpec, "://") {
+				return err
+			}
```

Comment:
What is this? If it is new code that you added because if you think it is necessary, I suppose it should have some comments.

If this is copy&pasted from elsewhere else in Buildah, it shouldn’t be copy&pasted; use a shared utility function to turn user input into an `ImageReference`. (I wouldn’t be surprised to heard that one — or multiple! — already exist, I didn’t go looking.)

Reply:
You're right, I got this piece from cmd/buildah/push.go. Based on your comment here and https://github.com/containers/buildah/pull/4677#discussion_r1174183375, I've moved it to util/util.go in order to reuse it in the both the places.